doc/src/hexdoc_hexcasting/_templates/render.js.jinja

```diff
@@ -0,0 +1,371 @@
+    pat.remove()
+  }
+}
+
+//last_load_func allows the previous event listeners to be removed
+//that way we aren't doing unecessary work when you switch between a ton of options
+let last_load_func = null;
+
+//lazily renders all of the images with the chosen render options
+//loads all of the currently open ones and then adds event listeners to open the rest when they're opened
+function render_images(draw_options) {
+  let collapsibles = document.getElementsByClassName("details-collapsible");
+
+  let load_func = (event) => {
+    render_collapsible(draw_options, event.target);
+  }
+  for (let collapsible of collapsibles) {
+    // @ts-ignore
+    collapsible.removeEventListener("toggle", last_load_func, {once: true});
+
+    if (collapsible.hasAttribute("open")) {
+      render_collapsible(draw_options, collapsible);
+    } else {
+      collapsible.addEventListener("toggle", load_func, {once: true});
+    }
+  }
+
+  last_load_func = load_func;
+}
+
+//loads all of the animated canvas elements back
+//also adds an event listener to enable them when their menu is opened
+function load_animated() {
+  let collapsibles = document.getElementsByClassName("details-collapsible");
+
+  let load_func = (event) => {
+    const canvases = event.target.querySelectorAll("canvas");
+    canvases.forEach(initializeElem);
+  };
+
+  for (let collapsible of collapsibles) {
+    let patterns = Array.from(collapsible.getElementsByClassName("spell-viz"));
+
+    let open = collapsible.hasAttribute("open");
+
+    // @ts-ignore
+    collapsible.removeEventListener("toggle", last_load_func, {once: true});
+
+    if (!open) {
+      collapsible.addEventListener("toggle", load_func, {once:true})
+    }
+    
+    for (let pattern of patterns) {
+      let attr_to_transfer = ["data-start", "data-string", "width", "height", "data-per-world", "class"];
+      let canvas = document.createElement("canvas");
+      for (const index in attr_to_transfer) {
+        const attr = attr_to_transfer[index];
+        let value = pattern.getAttribute(attr);
+        canvas.setAttribute(attr, value ? value : "");
+      }
+
+      canvas.innerHTML = pattern.innerHTML;
+
+      collapsible.appendChild(canvas);
+
+      if (open) {
+        initializeElem(canvas);
+      }
+
+      pattern.remove()
+    }
+  }
+
+  last_load_func = load_func;
+
+}
+
+/** @type {Point} */
+const intersection_point = {
+  type: "Single",
+  marker: {
+    color: [255, 255, 255, 150],
+    radius: 0.07,
+  }
+};
+
+/** @type {Intersections} */
+const intersections = {
+  type: "EndsAndMiddle",
+  start: {
+    type: "BorderedMatch",
+    match_radius: 0.04,
+    border: intersection_point.marker
+  },
+  middle: intersection_point,
+  end: {
+    type: "Point",
+    point: intersection_point
+  }
+};
+
+const line_thickness = 0.12;
+
+/** @type {Color[]} */
+const colors = [
+  [214, 9, 177, 255],
+  [108, 25, 140, 255],
+  [50, 102, 207, 255],
+  [102, 110, 125, 255],
```

Comment:
HexBug palette:
```suggestion
  [255, 107, 255, 255],
  [168, 30, 227, 255],
  [100, 144, 237, 255],
  [177, 137, 199, 255],
```

Reply:
I forgot to stage this with the rest of the palette fixes, so I'm putting it in manually.